packages/wonder-blocks-modal/components/modal-launcher.test.js

```diff
@@ -59,4 +59,34 @@ describe("ModalLauncher", () => {
         wrapper.find("button").simulate("click");
         expect(wrapper.find("ModalLauncherPortal")).toHaveLength(1);
     });
+
+    test("Pressing Escape closes the modal", (done) => {
+        let pressedEscape = false;
+
+        // On close, assert that this function was triggered _after_ Escape was
+        // pressed, and finish the test.
+        const onClose = () => {
+            expect(pressedEscape).toBe(true);
```

Comment:
Verify this isn't all that important.  I'd rather verify that the modal is gone which we can do since we `mount`ed the `ModalLauncher` for real.

Reply:
Yeah, I punted on that assertion, since it's a bit more complex to test. I suppose we can use a `ref` on the modal again in order to listen for mounts and unmounts, and keep that stored in a bool. I'll try that.

Reply:
It's easy to assert that the modal is not present in the DOM via `querySelector`… but it's not a very helpful assertion if we don't also assert at some point that the modal _was_ in the DOM, and there's not an obvious hook for when to do that :(

Reply:
I suppose it depends on the value proposition of asserting the modal's presence in the DOM. I don't think it actually simplifies the test, because I think we have to hook into lifecycle methods awkwardly in order to do that robustly. And we have other tests that assert that `onClose` is only called once the modal content will be unmounted, so I'm not worried about asserting that even more.

I'm also now starting to wonder whether a full mount is actually necessary here. I think Enzyme's shallow rendering _does_ trigger React lifecycle methods, so the event listeners will get registered anyway… maybe I should try that and see how it goes!

Reply:
Hmm, thinking more about it, I think we should just punt the async rendering concerns. Fingers crossed that, once async rendering lands, we'll still be able to use synchronous rendering for unit tests!

Reply:
Yeah, let's punt on them for now.  I hope we don't wait for async rendering in React before we have some more e2e tests for these components.